load-generator/src/main/java/com/opensearchloadtester/loadgenerator/service/QueryExecutionTask.java

```diff
@@ -50,21 +49,49 @@ public void run() {
             Response response = openSearchClient.execute(request);
             long requestDurationMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
 
+            int status = response.getStatus();
+
+            if (status >= 400) {
+                MetricsDto metricsDto = new MetricsDto(
+                        loadGeneratorId,
+                        queryType.name(),
+                        requestDurationMillis,
+                        null,
+                        null,
+                        status
+                );
+
+                metricsCollector.appendMetrics(metricsDto);
```

Comment:
Warte sollte das metricsDto nicht unabhängig von einem Fehler erstellt werden? Also >400 == Fehler. So hätten wir dann in einem Report ja nur erfolgreiche status.